output is sensitive to terminal width

rustc: 1.93.0 stable
trybuild: 1.0.114

----

Using the [PyO3 test suite](https://github.com/PyO3/pyo3/tree/main/tests/ui) as an example, I observe differences in output recorded by trybuild according to terminal width.

If I run the PyO3 test suite with a terminal around a default size of ~100 chars, the test suite passes.

If I resize the terminal to ~60 characters, and retry, I get failures such as

```
EXPECTED:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0597]: `local_data` does not live long enough
  --> tests/ui/invalid_closure.rs:7:27
   |
 6 |         let local_data = vec![0, 1, 2, 3, 4];
   |             ---------- binding `local_data` declared here
 7 |         let ref_: &[u8] = &local_data;
   |                           ^^^^^^^^^^^ borrowed value does not live long enough
...
14 |         PyCFunction::new_closure(py, None, None, closure_fn)
   |         ---------------------------------------------------- argument requires that `local_data` is borrowed for `'static`
...
17 |     });
   |     - `local_data` dropped here while still borrowed
   |
note: requirement that the value outlives `'static` introduced here
  --> src/types/function.rs
   |
   |         F: Fn(&Bound<'_, PyTuple>, Option<&Bound<'_, PyDict>>) -> R + Send + 'static,
   |                                                                              ^^^^^^^
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈

ACTUAL OUTPUT:
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
error[E0597]: `local_data` does not live long enough
  --> tests/ui/invalid_closure.rs:7:27
   |
 6 |         let local_data = vec![0, 1, 2, 3, 4];
   |             ---------- binding `local_data` declared here
 7 |         let ref_: &[u8] = &local_data;
   |                           ^^^^^^^^^^^ borrowed value does not live long enough
...
14 |         PyCFunction::new_closure(py, None, None, closure_fn)
   |         ---------------------------------------------------- argument requires that `local_data` is borrowed for `'static`
...
17 |     });
   |     - `local_data` dropped here while still borrowed
   |
note: requirement that the value outlives `'static` introduced here
  --> src/types/function.rs
   |
   | ...yDict>>) -> R + Send + 'static,
   |                           ^^^^^^^
┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈
```

Note the final line of output, for example, where the `F: Fn` bound has been truncated to match the small width of the host terminal.

--- a/src/cargo.rs
+++ b/src/cargo.rs
@@ -37,6 +37,10 @@ fn raw_cargo() -> Command {
 }
 
 fn cargo(project: &Project) -> Command {
+    cargo_with_rustflags(project, &[])
+}
+
+fn cargo_with_rustflags(project: &Project, extra_rustflags: &[&'static str]) -> Command {
     let mut cmd = raw_cargo();
     cmd.current_dir(&project.dir);
     cmd.envs(cargo_target_dir(project));
@@ -44,7 +48,7 @@ fn cargo(project: &Project) -> Command {
     cmd.env("CARGO_INCREMENTAL", "0");
     cmd.arg("--offline");
 
-    let rustflags = rustflags::toml();
+    let rustflags = rustflags::toml(extra_rustflags);
     cmd.arg(format!("--config=build.rustflags={rustflags}"));
     cmd.arg(format!("--config=target.{TARGET}.rustflags={rustflags}"));
 
@@ -122,7 +126,7 @@ pub(crate) fn build_test(project: &Project, name: &Name) -> Result<Output> {
         .stderr(Stdio::null())
         .status();
 
-    cargo(project)
+    cargo_with_rustflags(project, &["--diagnostic-width=140"])
         .arg(if project.has_pass { "build" } else { "check" })
         .args(target())
         .arg("--bin")
@@ -145,7 +149,7 @@ pub(crate) fn build_all_tests(project: &Project) -> Result<Output> {
         .stderr(Stdio::null())
         .status();
 
-    cargo(project)
+    cargo_with_rustflags(project, &["--diagnostic-width=140"])
         .arg(if project.has_pass { "build" } else { "check" })
         .args(target())
         .arg("--bins")

--- a/src/rustflags.rs
+++ b/src/rustflags.rs
@@ -2,7 +2,7 @@ use std::env;
 
 const IGNORED_LINTS: &[&str] = &["dead_code"];
 
-pub(crate) fn toml() -> toml::Value {
+pub(crate) fn toml(extra_rustflags: &[&'static str]) -> toml::Value {
     let mut rustflags = vec!["--cfg", "trybuild", "--verbose"];
 
     for &lint in IGNORED_LINTS {
@@ -17,6 +17,8 @@ pub(crate) fn toml() -> toml::Value {
             rustflags.extend(["-C", "instrument-coverage"]);
         }
     }
+
+    rustflags.extend(extra_rustflags);
 
     toml::Value::try_from(rustflags).unwrap()
 }

